pkg/3scale/amp/operator/apicast_options_provider.go

```diff
@@ -202,3 +209,119 @@ func (a *ApicastOptionsProvider) validateCustomPolicySecret(name string) error {
 
 	return nil
 }
+
+func (a *ApicastOptionsProvider) setTracingConfiguration() error {
+	err := a.setProductionTracingConfiguration()
+	if err != nil {
+		return err
+	}
+
+	err = a.setStagingTracingConfiguration()
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (a *ApicastOptionsProvider) setProductionTracingConfiguration() error {
+	tracingIsEnabled := a.apimanager.IsAPIcastProductionOpenTracingEnabled()
+	res := &component.APIcastTracingConfig{
+		Enabled:        tracingIsEnabled,
+		TracingLibrary: component.APIcastDefaultTracingLibrary,
+	}
+	if tracingIsEnabled {
+		openTracingConfigSpec := a.apimanager.Spec.Apicast.ProductionSpec.OpenTracing
+		if openTracingConfigSpec.TracingLibrary != nil {
+			// For now only "jaeger" is accepted" as the tracing library
```

Comment:
This check seems to belong to APIManager CR `Validate()` method. IT is being validated before trying reconciliation. The option provider is executed as part of the reconciliation.

Reply:
changed :+1: